### PR TITLE
Move (most) Debugger Attributes to shared partition

### PR DESCRIPTION
**Pull request by benaadams**
corert change https://github.com/dotnet/corert/pull/4774

### CODE REVIEW

File: src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs [hunk 1]
```diff
@@ -22,239 +21,6 @@ public sealed class DebuggerStepThroughAttribute : Attribute
     {
         public DebuggerStepThroughAttribute() { }
```
Comment (jkotas) [inline src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs]: I would move all of them, and add `[System.Runtime.CompilerServices.DependencyReductionRoot]` under `#if !PROJECTN` ifdef.

Comment (benaadams) [reply src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs]: Is `PROJECTN` coreclr?

Comment (jkotas) [reply src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs]: PROJECTN means .NET Native. This attribute is only used by the (closed source) .NET Native toolchain.

Comment (benaadams) [reply src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs]: So `#if PROJECTN` rather than `#if !PROJECTN`?

**Comment by jkotas** [comment]
Maybe even move the two debugger attributes from https://github.com/dotnet/corefx/tree/master/src/System.Diagnostics.Debug/src/System/Diagnostics back to CoreLib. It does not really make sense to have all debugger attributes in CoreLib, except two that happened to not be used in CoreLib.

File: src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs (hunk 1 above)
Comment (jkotas) [reply src/mscorlib/src/System/Diagnostics/DebuggerAttributes.cs]: Right.

**Comment by jkotas** [comment]
Superseded by https://github.com/dotnet/coreclr/pull/15100

**Pull request closed without merging**